apps/avatax/src/modules/saleor/shipping-line.ts

```diff
@@ -1,39 +1,90 @@
-import { LineItemModel } from "avatax/lib/models/LineItemModel";
+import { Result, err, ok } from "neverthrow";
+import { z } from "zod";
+import { BaseError } from "../../error";
 import { SHIPPING_ITEM_CODE } from "../avatax/calculate-taxes/avatax-shipping-line";
 
-export class SaleorShippingLine {
-  private getAmount({
-    gross,
-    net,
-    taxIncluded,
-  }: {
-    gross: number;
-    net: number;
-    taxIncluded: boolean;
-  }) {
-    return taxIncluded ? gross : net;
+export interface ISaleorOrderConfirmedShippingLine {
```

Comment:
I'm not sure if we need interface here. There is a difference when class is a service and value object.
Service has interface, because it is dependency of something. This "Something" *requires* interface (behavior) and we can use something else that implements this interface (mock, but also some class that extends other class etc).

With value object, we have *business representation* here. Yes, it contains behavior too. But its *implementation of the interface* that guarantees proper object value (interface doesn't parse/validate anything, class does).

So every time you ALLOW to accept interface, you allow to omit validation/parsing. I think it should be other way around:
1. We should create very strict domain objects that radiate their business capabilities
2. We should change services to accept only this specific value objects / entities implementations, to ensure they were validated before

This way in services we know that we operate on correct data